src/Microsoft.DotNet.Build.Tasks/GenerateTestExecutionScripts.cs

```diff
@@ -121,7 +121,16 @@ private void WriteCmdExecutionScript(string cmdExecutionTemplate, string outputP
                 // Every test project comes with 4 of them, so not producing a warning here.
                 else if (!string.IsNullOrEmpty(relativePath))
                 {
-                    copyCommands.AppendLine($"call :copyandcheck \"%PACKAGE_DIR%\\{relativePath}\" \"%EXECUTION_DIR%\\{Path.GetFileName(relativePath)}\" ||  exit /b -1");
+                    bool? preserveSubDirectories = dependency.GetMetadata("PreserveSubDirectories")?.Equals("true", StringComparison.OrdinalIgnoreCase);
+                    string filePath = Path.GetFileName(relativePath);
+                    if (preserveSubDirectories == true)
```

Comment:
I'm pretty sure there's already a metadata item that contains what you're calculating here...